### PR TITLE
add WarezCDN

### PR DESCRIPTION
**Pull request by lonelil**
This pull request resolves #XXX

 - [x] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [x] I have tested all of my changes.

### CODE REVIEW

File: src/providers/sources/warezcdn/index.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  embedsHost.forEach(async (element) => {
+    const embedHost = $(element).attr('data-load-embed-host')!;
+    const embedUrl = $(element).attr('data-load-embed')!;
+
+    if (embedHost === 'mixdrop') {
+      const params = new URLSearchParams({
+        id: embedUrl,
+        sv: 'mixdrop',
+      });
+      const realUrl = await ctx.proxiedFetcher<string>(`/getPlay.php?${params}`, {
```
Comment (JorrinKievit) [inline src/providers/sources/warezcdn/index.ts:37]: Imo more constistent to use query object here, so 
```
{
  baseUrl: warezcdnApiBase,
  query: {
      id: embedUrl,
     sv: 'mixdrop'
}
```

File: src/providers/sources/warezcdn/index.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          embedId: warezcdnembedMp4Scraper.id,
+          url: embedUrl,
+        },
+      );
+    }
+  });
+
+  return {
+    embeds,
+  } as SourcererOutput;
```
Comment (JorrinKievit) [inline src/providers/sources/warezcdn/index.ts:66]: Hmm why is the type assertion needed 🤔

File: src/providers/embeds/warezcdn/common.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  if (!stream.id) throw new NotFoundError("can't get stream id");
+
+  const decryptedId = decrypt(stream.id);
+
+  if (!decryptedId) throw new NotFoundError("can't get file id");
+
+  return decryptedId;
+}
+
+export const warezcdnWorkerProxy = 'https://workerproxy.warezcdn.workers.dev';
```
Comment (JorrinKievit) [inline src/providers/embeds/warezcdn/common.ts:57]: can this be moved to the common.ts file with the other constants

File: src/providers/embeds/warezcdn/common.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  // Remove the last 5 characters from the original string
+  output = output.slice(0, -5);
+
+  // Return the original string concatenated with the reversed last 5 characters
+  return `${output}${last}`;
+}
+
+export async function getDecryptedId(ctx: EmbedScrapeContext) {
+  const page = await ctx.proxiedFetcher<string>(`/player.php?${new URLSearchParams({ id: ctx.url })}`, {
```
Comment (JorrinKievit) [inline src/providers/embeds/warezcdn/common.ts:26]: Same here with the query object

**Review by JorrinKievit — CHANGES_REQUESTED** [review]

File: src/providers/sources/warezcdn/index.ts (hunk 1 above)
Comment (lonelil) [reply src/providers/sources/warezcdn/index.ts:37]: omg i forgot the query option existed

File: src/providers/sources/warezcdn/index.ts (hunk 2 above)
Comment (lonelil) [reply src/providers/sources/warezcdn/index.ts:66]: was copying from my own project, forgot to remove

**Comment by ztpn** [comment]
> also the ranks has to be adjusted

yes, hls should be ramked above mp4
and add CORS_ALLOWED to mp4 and the source scraper

**Review by JorrinKievit — APPROVED** [review]

**Pull request merged**